### PR TITLE
Run the upgrade and downgrade tests in the upgrade prow job

### LINKED ISSUE #310: Run the upgrade and downgrade tests in serving repo
**Describe the bug**
Operator installs the knative operand, but it does not run the upgrade and downgrade tests available for knative.

**Expected behavior**
In the upgrade prow job, we need to add the upgrade and downgrade tests, verifying the knative serving.

**To Reproduce**
Steps to reproduce the behavior.
Submit a PR. Upgrade prow job is kicked off.

**Knative release version**
HEAD

**Additional context**
Add any other context about the problem here such as proposed priority

### PR DESCRIPTION
**Pull request by houshengbo**
Fixes #310

## Proposed Changes

* This PR adds the execution of upgrade and downgrade tests into the operator prow job.
* This PR refactors the e2e-common script, to accommodate the changes in upgrade tests.
* This PR adds the deployment verification for Knative in operator's upgrade tests. 

**Release Note**

```release-note

```

### CODE REVIEW

File: test/upgrade/servingoperator_preupgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
+//// +build preupgrade
+
+/*
+Copyright 2018 The Knative Authors
```
Comment (mattmoor) [inline test/upgrade/servingoperator_preupgrade_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2018 The Knative Authors
```
Comment (mattmoor) [inline test/upgrade/upgrade.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/servingoperator_preupgrade_test.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
+// +build preupgrade
+
+/*
+Copyright 2018 The Knative Authors
```
Comment (mattmoor) [inline test/upgrade/servingoperator_preupgrade_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/upgrade.go (hunk 2 above)
Comment (mattmoor) [inline test/upgrade/upgrade.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/servingoperator_preupgrade_test.go (hunk 3 above)
Comment (mattmoor) [inline test/upgrade/servingoperator_preupgrade_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/upgrade.go (hunk 2 above)
Comment (mattmoor) [inline test/upgrade/upgrade.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/e2e-common.sh [hunk 4]
```diff
@@ -97,3 +132,17 @@ function knative_teardown() {
...
+  ko apply ${KO_FLAGS} -f test/configuration/ || return 1
+  if (( MESH )); then
+    kubectl label namespace serving-tests istio-injection=enabled
+    kubectl label namespace serving-tests-alt istio-injection=enabled
+    ko apply ${KO_FLAGS} -f test/configuration/mtls/ || return 1
+  fi
+
+  echo ">> Uploading test images..."
+  ${REPO_ROOT_DIR}/test/upload-test-images.sh || return 1
+}
```
Comment (mattmoor) [inline test/e2e-common.sh]: Add trailing newline:
```suggestion
}

```

Comment (mattmoor) [inline test/e2e-common.sh]: Add trailing newline:
```suggestion
}

```

File: test/resources/service.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline test/resources/service.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/test_images/pizzaplanetv1/main.go [hunk 6]
```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2018 The Knative Authors
```
Comment (mattmoor) [inline test/test_images/pizzaplanetv1/main.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/test_images/pizzaplanetv2/main.go [hunk 7]
```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2018 The Knative Authors
```
Comment (mattmoor) [inline test/test_images/pizzaplanetv2/main.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/probe_test.go [hunk 8]
```diff
@@ -0,0 +1,71 @@
+// +build probe
+
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline test/upgrade/probe_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/service_postupgrade_test.go [hunk 9]
```diff
@@ -0,0 +1,104 @@
+// +build postupgrade1
+
+/*
+Copyright 2018 The Knative Authors
```
Comment (mattmoor) [inline test/upgrade/service_postupgrade_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/servingoperator_preupgrade_test.go [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
```
Comment (mattmoor) [inline test/upgrade/servingoperator_preupgrade_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/upgrade.go (hunk 2 above)
Comment (mattmoor) [inline test/upgrade/upgrade.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/resources/service.go (hunk 5 above)
Comment (mattmoor) [inline test/resources/service.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/test_images/pizzaplanetv1/main.go (hunk 6 above)
Comment (mattmoor) [inline test/test_images/pizzaplanetv1/main.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/test_images/pizzaplanetv2/main.go (hunk 7 above)
Comment (mattmoor) [inline test/test_images/pizzaplanetv2/main.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/probe_test.go (hunk 8 above)
Comment (mattmoor) [inline test/upgrade/probe_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/service_postupgrade_test.go (hunk 9 above)
Comment (mattmoor) [inline test/upgrade/service_postupgrade_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/servingoperator_preupgrade_test.go (hunk 10 above)
Comment (mattmoor) [inline test/upgrade/servingoperator_preupgrade_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/upgrade.go (hunk 2 above)
Comment (mattmoor) [inline test/upgrade/upgrade.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/resources/service.go (hunk 5 above)
Comment (mattmoor) [inline test/resources/service.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/test_images/pizzaplanetv1/main.go (hunk 6 above)
Comment (mattmoor) [inline test/test_images/pizzaplanetv1/main.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/test_images/pizzaplanetv2/main.go (hunk 7 above)
Comment (mattmoor) [inline test/test_images/pizzaplanetv2/main.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/probe_test.go (hunk 8 above)
Comment (mattmoor) [inline test/upgrade/probe_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/service_postupgrade_test.go (hunk 9 above)
Comment (mattmoor) [inline test/upgrade/service_postupgrade_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/servingoperator_preupgrade_test.go (hunk 10 above)
Comment (mattmoor) [inline test/upgrade/servingoperator_preupgrade_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/upgrade.go (hunk 2 above)
Comment (mattmoor) [inline test/upgrade/upgrade.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/resources/service.go (hunk 5 above)
Comment (mattmoor) [inline test/resources/service.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/test_images/pizzaplanetv1/main.go (hunk 6 above)
Comment (mattmoor) [inline test/test_images/pizzaplanetv1/main.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/test_images/pizzaplanetv2/main.go (hunk 7 above)
Comment (mattmoor) [inline test/test_images/pizzaplanetv2/main.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/probe_test.go (hunk 8 above)
Comment (mattmoor) [inline test/upgrade/probe_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/service_postupgrade_test.go (hunk 9 above)
Comment (mattmoor) [inline test/upgrade/service_postupgrade_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/servingoperator_preupgrade_test.go (hunk 10 above)
Comment (mattmoor) [inline test/upgrade/servingoperator_preupgrade_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/upgrade/upgrade.go (hunk 2 above)
Comment (mattmoor) [inline test/upgrade/upgrade.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/e2e-common.sh [hunk 11]
```diff
@@ -17,16 +17,20 @@
...
+LATEST_SERVING_RELEASE_VERSION="v0.12.1"
 # Istio version we test with
 readonly ISTIO_VERSION="1.4.2"
 # Test without Istio mesh enabled
 readonly ISTIO_MESH=0
 # Namespace used for tests
-readonly TEST_NAMESPACE="operator-tests"
+readonly TEST_NAMESPACE="knative-serving"
+
+OPERATOR_DIR=$(dirname $0)/..
```
Comment (aliok) [inline test/e2e-common.sh]:
```suggestion
OPERATOR_DIR=$(cd $(dirname "$0")/.. && pwd)
```

`OPERATOR_DIR=$(dirname $0)/..` resolves to `./..`. 
Changing it with `OPERATOR_DIR=$(cd $(dirname "$0")/.. && pwd)` results in `OPERATOR_DIR` being e.g. `/home/ali/go/src/knative.dev/`

**Comment by aliok** [comment]
@houshengbo any instructions to run this test locally? I couldn't on Minikube and on OpenShift

**Comment by houshengbo** [comment]
@aliok this is good question, but it seems to me that running the tests against GKE is the ONLY way to verify the tests.

**Comment by aliok** [comment]
> but it seems to me that running the tests against GKE is the ONLY way to verify the tests.

then it lgtm as things pass on GKE. I created https://github.com/knative/serving-operator/issues/317 so that I don't forget it and hopefully make things working later.

/lgtm

**Pull request merged**